website/addons/osfstorage/views.py

```diff
@@ -160,18 +163,21 @@ def osfstorage_create_child(file_node, payload, node_addon, **kwargs):
 
     if not is_folder:
         try:
-            version = file_node.create_version(
-                user,
-                dict(payload['settings'], **dict(
-                    payload['worker'], **{
-                        'object': payload['metadata']['name'],
-                        'service': payload['metadata']['provider'],
-                    })
-                ),
-                dict(payload['metadata'], **payload['hashes'])
-            )
-            version_id = version._id
-            archive_exists = version.archive is not None
+            if file_node.checkout is None or file_node.checkout == user:
+                version = file_node.create_version(
+                    user,
+                    dict(payload['settings'], **dict(
+                        payload['worker'], **{
+                            'object': payload['metadata']['name'],
+                            'service': payload['metadata']['provider'],
+                        })
+                    ),
+                    dict(payload['metadata'], **payload['hashes'])
+                )
+                version_id = version._id
+                archive_exists = version.archive is not None
+            else:
+                raise HTTPError(httplib.UNAUTHORIZED)
```

Comment:
401 doesn't seem appropriate here. `UNAUTHORIZED` means that the client has failed to provide valid authentication credentials. I think 403 (Forbidden) makes more sense. Also, add a useful `message_long` to this error.